### PR TITLE
lesson 2: lecture gist

### PR DESCRIPTION
**Pull request by stlaz**
Adding lesson 2 lecture, currently a rough gist

### CODE REVIEW

File: lesson_2/README.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
+%title: Introduction To Pods and Pod Controllers
+%author: Stanislav Láznička
```
Comment (TomasTomecek) [inline lesson_2/README.md]: this is fancy! is it some framework or a tool that we can use for visualization?

File: lesson_2/README.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+---
+
+## Service
+- represents a service, keeps stable hostname
+- <connecting a service to a pod>
+- <types of services and their use>
+
+---
+
+## Route
```
Comment (TomasTomecek) [inline lesson_2/README.md]: I got personally bit by Ingress vs. Route: could you please briefly mention it?

**Review by TomasTomecek — COMMENTED** [review]
looking good

File: lesson_2/README.md (hunk 1 above)
Comment (stlaz) [reply lesson_2/README.md]: Oh yes! Check out mdp, I was so happy when I found it: https://github.com/visit1985/mdp

File: lesson_2/README.md (hunk 2 above)
Comment (stlaz) [reply lesson_2/README.md]: Good idea, I'll add that in

File: lesson_2/README.md (hunk 1 above)
Comment (TomasTomecek) [reply lesson_2/README.md]: not packaged in Fedora :(

Comment (stlaz) [reply lesson_2/README.md]: It is in Arch, I think you could also use https://copr.fedorainfracloud.org/coprs/sentry/mdp/

File: lesson_2/README.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+## Pod
+
+- basic building stone of kubernetes
+- defines what to run and how to run it
+    - ports
+    - host ports
+    - volumes
+    - environment variables
+    - security attributes
+    - ...
```
Comment (TomasTomecek) [inline lesson_2/README.md]: please add container image :D

File: lesson_2/README.md [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+
+---
+
+## Pod
+
+- may consist of multiple containers
+- container types
+    - init containers
+    - normal containers
+    - new (1.23 - beta) - [ephemeral containers](https://k8s.io/docs/concepts/workloads/pods/ephemeral-containers/)
```
Comment (TomasTomecek) [inline lesson_2/README.md]: neat, didn't know about this

**Review by TomasTomecek — COMMENTED** [review]
looking pretty good! let me know if you need help with the demos :)

File: lesson_2/example/deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    type: Recreate
+  template:
+    metadata:
+      labels:
+        app: openshift-demo
+    spec:
+        imagePullPolicy: IfNotPresent
+        containers:
+        - name: httpbin
+          image: kennethreitz/httpbin:latest
```
Comment (TomasTomecek) [inline lesson_2/example/deployment.yaml]: very nice! this should be a convenient demo for all

File: lesson_2/README.md [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+- maintains order of pod deployment
+- API directly allows requesting PVs
+    - for stable storage access
+    - *spec.volumeClaimTemplates*
+
+---
+
+## StatefulSets Demo
+
+- [Deploying a replicated MySQL database](https://kubernetes.io/docs/tasks/run-application/run-replicated-stateful-application/)
```
Comment (TomasTomecek) [inline lesson_2/README.md]: I also like this demo - mysql is a great example

**Comment by stlaz** [comment]
we should probably merge this before the demo starts :slightly_smiling_face:

**Pull request merged**